packages/cactus-test-tooling/src/main/typescript/postgres/postgres-test-container.ts

```diff
@@ -0,0 +1,329 @@
+    }
+    this.imageVersion =
+      options.imageVersion ||
+      POSTGRES_TEST_CONTAINER_DEFAULT_OPTIONS.imageVersion;
+    this.imageName =
+      options.imageName || POSTGRES_TEST_CONTAINER_DEFAULT_OPTIONS.imageName;
+    this.postgresPort =
+      options.postgresPort ||
+      POSTGRES_TEST_CONTAINER_DEFAULT_OPTIONS.postgresPort;
+    this.envVars =
+      options.envVars || POSTGRES_TEST_CONTAINER_DEFAULT_OPTIONS.envVars;
+
+    this.emitContainerLogs = Bools.isBooleanStrict(options.emitContainerLogs)
+      ? (options.emitContainerLogs as boolean)
+      : true;
+
+    this.validateConstructorOptions();
+    const label = "postgres-test-container";
+    const level = options.logLevel || "INFO";
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+
+  public getContainer(): Container {
+    const fnTag = "PostgresTestContainer#getContainer()";
+    if (!this.container) {
+      throw new Error(`${fnTag} container not yet started by this instance.`);
+    } else {
+      return this.container;
+    }
+  }
+
+  public getimageName(): string {
+    return `${this.imageName}:${this.imageVersion}`;
+  }
+
+  public async getPostgresPortHost(): Promise<string> {
+    const ipAddress = "127.0.0.1";
+    const hostPort: number = await this.getPostgresPort();
+    return `http://${ipAddress}:${hostPort}`;
+  }
+
+  public async getFileContents(filePath: string): Promise<string> {
+    const response: any = await this.getContainer().getArchive({
+      path: filePath,
+    });
+    const extract: tar.Extract = tar.extract({ autoDestroy: true });
+
+    return new Promise((resolve, reject) => {
+      let fileContents = "";
+      extract.on("entry", async (header: any, stream, next) => {
+        stream.on("error", (err: Error) => {
+          reject(err);
+        });
+        const chunks: string[] = await Streams.aggregate<string>(stream);
+        fileContents += chunks.join("");
+        stream.resume();
+        next();
+      });
+
+      extract.on("finish", () => {
+        resolve(fileContents);
+      });
+
+      response.pipe(extract);
+    });
+  }
+
+  public async start(): Promise<Container> {
+    const imageFqn = this.getimageName();
+
+    if (this.container) {
+      await this.container.stop();
+      await this.container.remove();
+    }
+    const docker = new Docker();
+
+    this.log.debug(`Pulling container image ${imageFqn} ...`);
+    await this.pullContainerImage(imageFqn);
+    this.log.debug(`Pulled ${imageFqn} OK. Starting container...`);
+
+    return new Promise<Container>((resolve, reject) => {
+      const eventEmitter: EventEmitter = docker.run(
+        imageFqn,
+        [],
+        [],
+        {
+          Env: this.envVars,
+          Healthcheck: {
+            Test: ["CMD-SHELL", "pg_isready -U postgres"],
+            Interval: 1000000000, // 1 second
+            Timeout: 3000000000, // 3 seconds
+            Retries: 299,
+            StartPeriod: 3000000000, // 1 second
```

Comment:
as above:)

Reply:
You are correct! Thank you for pointing out! Fixed.